apps/omg_eth/lib/omg_eth/ethereum_height_monitor.ex

```diff
@@ -0,0 +1,150 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Eth.EthereumHeightMonitor do
```

Comment:
hmmm wouldn't it be easier if this was done by `EthereumClientMonitor`? It's kinda has that purpose.

Reply:
so if you did that, you'd only need to send yourself a :check interval every 20seconds and check if the ethereum block height has increased, if it didn't you'd kill the linked process.

Reply:
Yeah intentionally copied from `EthereumClientMonitor`. Mainly because adding it would expand `EthereumClientMonitor`'s states & branching quite a bit:

- 2 intervals
- 2 timers: 1 for client health check, 1 for height check
- 2 raised alarm tracking
- 2 heights tracking: ethereum_height and previous ethereum_height

Up for suggestions though!

Reply:
> so if you did that, you'd only need to send yourself a :check interval every 20seconds and check if the ethereum block height has increased, if it didn't you'd kill the linked process.

👌 Lemme code it up quickly so we can compare

Reply:
since we're stripping down WS connections this might become his only work...
and maybe doing https://github.com/omisego/elixir-omg/issues/1255 this first might be easier or less tech debt? just thinking out loud...